src/core/Akka.Remote/MessageSerializer.cs

```diff
@@ -39,19 +40,18 @@ public static object Deserialize(ExtendedActorSystem system, SerializedMessage m
         /// Serializes the specified message.
         /// </summary>
         /// <param name="system">The system.</param>
-        /// <param name="address">TBD</param>
+        /// <param name="transportInformation">The address for the current transport</param>
         /// <param name="message">The message.</param>
         /// <returns>SerializedMessage.</returns>
-        public static SerializedMessage Serialize(ExtendedActorSystem system, Address address, object message)
+        public static SerializedMessage Serialize(ExtendedActorSystem system, Information transportInformation,
+            object message)
         {
             var serializer = system.Serialization.FindSerializerFor(message);
 
             var oldInfo = Akka.Serialization.Serialization.CurrentTransportInformation;
             try
             {
-                if (oldInfo == null)
-                    Akka.Serialization.Serialization.CurrentTransportInformation =
-                        system.Provider.SerializationInformation;
+                Akka.Serialization.Serialization.CurrentTransportInformation = transportInformation;
```

Comment:
The other part of the real fix - make sure we explicitly set the `CurrentTransportInformation` to whatever was passed in by the `EndpointReader` before we get going